python/ray/serve/_private/proxy_response_generator.py

```diff
@@ -0,0 +1,147 @@
+import asyncio
+import logging
+import time
+from abc import ABC, abstractmethod
+from asyncio.tasks import FIRST_COMPLETED
+from typing import Any, Callable, Optional, Union
+
+from ray.serve._private.constants import SERVE_LOGGER_NAME
+from ray.serve._private.utils import calculate_remaining_timeout
+from ray.serve.handle import DeploymentResponse, DeploymentResponseGenerator
+
+logger = logging.getLogger(SERVE_LOGGER_NAME)
+
+
+class _ProxyResponseGeneratorBase(ABC):
+    def __init__(
+        self,
+        *,
+        timeout_s: Optional[float] = None,
+        disconnected_task: Optional[asyncio.Task] = None,
+        result_callback: Optional[Callable[[Any], Any]] = None,
+    ):
+        """Implements a generator wrapping a deployment response.
+
+        Args:
+            - timeout_s: an end-to-end timeout for the request. If this expires and the
```

Comment:
nit: Add "None" value behavior for timeout_s.

Reply:
added